lib/tapioca/helpers/rbi_files_helper.rb

```diff
@@ -175,7 +175,15 @@ def shims_or_todos_have_duplicates?(nodes, shim_rbi_dir:, todo_rbi_file:)
       return false if shims_or_todos.empty?
 
       shims_or_todos_empty_scopes = extract_empty_scopes(shims_or_todos)
-      return true unless shims_or_todos_empty_scopes.empty?
+      shims_or_todos_empty_scopes.each do |shim_or_todo|
+        return true unless shim_or_todo.is_a?(RBI::Class)
+
+        nodes.each do |node|
+          return true unless node.is_a?(RBI::Class)
+          next if node == shim_or_todo
+          return true if node.superclass_name == shim_or_todo.superclass_name
+        end
+      end
```

Comment:
Can this all not be:
```suggestion
      return true if shims_or_todos_empty_scopes.grep(RBI::Class).any? do |shim_or_todo|
        nodes.grep(RBI::Class).any? do |node|
          node != shim_or_todo && node.superclass_name == shim_or_todo.superclass_name
        end
      end
```
which can then be encapsulated in a method.